api/static/swagger.yaml

```diff
@@ -173,6 +190,96 @@ paths:
         204:
           description: "No content"
 
+  "/v1/projects/{project_id}/secret_storages":
+    post:
+      tags: ["secret_storage"]
+      summary: "Create secret storage"
+      parameters:
+        - in: "path"
+          name: "project_id"
+          type: "integer"
+          required: true
+        - in: "body"
+          name: "body"
+          required: true
+          schema:
+            $ref: "#/definitions/SecretStorage"
+      responses:
+        201:
+          description: "Created"
+          schema:
+            $ref: "#/definitions/SecretStorage"
+    get:
+      tags: ["secret_storage"]
+      summary: "List secret storage"
+      parameters:
+        - in: "path"
+          name: "project_id"
+          type: "integer"
+          required: true
+      responses:
+        200:
+          description: "Ok"
+          schema:
+            type: "array"
+            items:
+              $ref: "#/definitions/SecretStorage"
+
+  "/v1/projects/{project_id}/secret_storages/{secret_storage_id}":
+    get:
+      tags: ["secret_storage"]
+      summary: "Get secret storage"
+      parameters:
+        - in: "path"
+          name: "project_id"
+          type: "integer"
+          required: true
+        - in: "path"
+          name: "secret_storage_id"
+          type: "integer"
+          required: true
+      responses:
+        200:
+          description: "Ok"
+          schema:
+            $ref: "#/definitions/SecretStorage"
+    patch:
```

Comment:
Is `put` a better verb since we're allowing replacing the whole entity (minus the id) ?

Reply:
I choose `patch` for consistency with existing [update secret API](https://github.com/caraml-dev/mlp/blob/54d1b5c6076e6b7645a54f5874b2aeb47cad385a/api/static/swagger.yaml#L156). Additionally, the implementation allow update request with [partial request body ](https://github.com/caraml-dev/mlp/blob/54d1b5c6076e6b7645a54f5874b2aeb47cad385a/api/api/secret_storages_api_test.go#L377)which is more consistent with `patch` behavior.